include/vcpkg/base/diagnostics.h

```diff
@@ -68,6 +68,8 @@ namespace vcpkg
         DiagnosticLine reduce_to_warning() const&;
         DiagnosticLine reduce_to_warning() &&;
 
+        const LocalizedString& message_text() const noexcept { return m_message; }
```

Comment:
I kind of don't like exposing the message alone like this but the transform JsonReader wants to do kinda needs it. I suspect we don't have the abstractions we should be using quite right in there but I wanted to do something I could practically do now rather than trying to redo the whole JsonReader/ParserBase world right now.